finetune/base_models/gpt2/encoder.py

```diff
@@ -175,36 +181,32 @@ def _encode(self, texts, labels=None):
                 subtoken_idxs.extend(
                     [self.encoder.get(t, self.UNK_IDX) for t in bpe_toks]
                 )
-
-                token_char_starts = [token_start] * len(bpe_toks)
-
-                if np.sum([len(tok) for tok in bpe_toks]) > len(token):
-                    token_char_ends = (
-                        np.asarray([len(token.strip()) for tok in bpe_toks])
-                        + token_start
-                    )
-                else:
-                    token_char_ends = (
-                        np.cumsum([len(tok) for tok in bpe_toks]) + token_start
-                    )
-
+                lens = []
```

Comment:
Minor point but maybe pre-allocate this array to something to prevent resizing several times